### PR TITLE
Added a check for the asset tag int size

### PR DESCRIPTION
**Pull request by Godmartinz**
# Description
Adds a check to see if an `asset_tag` is above the `PHP_INT_MAX`. if so, you are redirected back to the asset edit page you were on with the following error message.

image

Fixes # [sc-25213]

## Type of change

Please delete options that are not relevant.

- [x ] Bug fix (non-breaking change which fixes an issue)

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Test A
- [ ] Test B

**Test Configuration**:
* PHP version:
* MySQL version
* Webserver version
* OS version

# Checklist:

- [ ] I have read the Contributing documentation available here: https://snipe-it.readme.io/docs/contributing-overview
- [ ] I have formatted this PR according to the project guidelines: https://snipe-it.readme.io/docs/contributing-overview#pull-request-guidelines
- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Comment by snipe** [comment]
@uberbrady I know you and I have talked about this a lot - is this the way we want to handle this? The error is a MySQL error, not a PHP error, so I'm not sure this is the best approach.

**Comment by uberbrady** [comment]
My "ridiculously complicated wonky solution" doesn't seem to be any better than this one - here's what I saw in Tinker:

```
>>> PHP_INT_MAX
=> 9223372036854775807

>>> PHP_INT_MAX > "9223372036854775808"
=> false

>>> PHP_INT_MAX > "922337203685477580800000000"
=> false

>>> PHP_INT_MAX > "9"
=> true

>>> PHP_INT_MAX > "9223372036854775807"
=> false

>>> PHP_INT_MAX > "9223372036854775808"
=> false

>>> PHP_INT_MAX > "9223372036854775806"
=> true

```

So I think this seems like a really solid way to solve the problem - nice work!

File: app/Http/Controllers/Assets/AssetsController.php [hunk 1]
```diff
@@ -111,6 +111,13 @@ public function store(ImageUploadRequest $request)
         // differently
         $asset_tags = $request->input('asset_tags');
 
+        foreach($asset_tags as $asset_tag) {
+            if ($asset_tag > PHP_INT_MAX) {
```
Comment (uberbrady) [inline app/Http/Controllers/Assets/AssetsController.php]: I think the only thing I might change here is to make this `>=` ? Because if you *do* try to save an asset with an asset tag of `PHP_INT_MAX` it's going to try to save that `+1` to the database - which will fail.

**Review by uberbrady — CHANGES_REQUESTED** [review]
One tiny change and this should be good to go. But, also, some tests here might be nice? We have some tests around asset tags and increments that you could work with, so you wouldn't be starting from zero. Me or @spencerrlongg or @marcusmoore can probably help you if you need to know where to start.

**Comment by Godmartinz** [comment]
I wrote a test, but unable to test it at the moment. Will check it in the morning.

**Comment by Godmartinz** [comment]
@uberbrady @snipe  So came to the realization that the way we currently store asset tags the prefix is built into the `asset_tag` column and it is _very_ hard to limit the size of this value this way.  The prefix could be numeric or string.

Moving forward if we store the prefix and the `asset_tag` into two separate columns we would be able to properly validate these things. In terms of search-ability if done this way we would utilize a virtual column to search upon--to not lose functionality. This realization is kind of a perfect segue into the buffing of asset tag system ticket. We could correct this problem moving forward.
